### PR TITLE
#14462 #14982 Setting unique template for working versions only

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -3530,12 +3530,20 @@ private void updateTemplateInAllLanguageVersions(final Contentlet contentlet, fi
...
+
+                                if (c.isWorking()){
+                                    if (!c.isLive()){
+                                        //Update all working versions of a page
+                                        updateContentletTemplate(c, newTemplate, fieldVar);
+                                    }else{
+                                        //Create a new working version with the template when the page version is live and working
+                                        Contentlet newPageVersion = checkout(c.getInode(), user, false);
+                                        newPageVersion.setStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD, newTemplate);
+                                        checkin(newPageVersion, user, false);
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: Maybe a DONT_VALIDATE_ME for the checked in content?

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 2]
```diff
@@ -3547,6 +3555,14 @@ public void run() {
         }
     }
 
+    private void updateContentletTemplate(Contentlet c, String newTemplate, Field fieldVar) throws DotDataException {
+        contentFactory.updateContentletTemplate(c.getInode(), newTemplate,
+                fieldVar.getFieldContentlet());
+        CacheLocator.getContentletCache().remove(c);
+        CacheLocator.getHTMLPageCache().remove(c.getInode());
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: I think the getHTMLPageCache takes an identifier?  Or is this correct?

**Review by dsilvam — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by wezell — COMMENTED** [review]
Please make sure the comments below are discussed.  I think the DO_NOT_VALIDATE needs to be implemented (thinking about pushed content or other cases).  I don't know about the html page cache.

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: Take a look at https://github.com/dotCMS/enterprise-2.x/blob/master/src/main/java/com/dotcms/enterprise/publishing/remote/handler/HandlerUtil.java#L114

**Pull request closed without merging**